pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -316,6 +312,10 @@ func (r *ReconcileNetworkAddonsConfig) Reconcile(request reconcile.Request) (rec
 	// perform the first check manually.
 	r.statusManager.SetFromPods()
 
+	if r.clusterInfo.MonitoringAvailable {
+		monitoring.TrackMonitoredComponents(&networkAddonsConfig.Spec, r.statusManager)
```

Comment:
Could you please explain what is the idea behind this change?

Reply:
the status can be updated on `r.statusManager.SetFromPods()` to available.

I moved the `TrackMonitoredComponents` to run after it, so that if the status is changed there - then we'll catch it more quickly.

Reply:
Now I get it, thanks!